README.md

```diff
@@ -1,6 +1,63 @@
-Alertmanager
-=============
+# Alertmanager
 
-Prometheus Alertmanager - still experimental!
+Prometheus Alertmanager - **still experimental!**
 
-See [config/fixtures/sample.conf.input](config/fixtures/sample.conf.input) for an example config.
+Alertmanager receives alerts generated by Prometheus and takes care of the
+following aspects:
+
+* manual silencing of specific alerts
+* inhibiting alerts based on alert dependencies
+* aggregating alerts by labelset
+* handling notification repeats
+* sending alert notifications via external services (currently email or [PagerDuty](http://www.pagerduty.com/))
+
+See [config/fixtures/sample.conf.input](config/fixtures/sample.conf.input) for
+an example config. The full configuration schema including a documentation for
+all possible options can be found in
+[config/config.proto](config/config.proto). Alertmanager automatically reloads
+the configuration when it changes, so restarts are not required for
+configuration updates.
+
+## Building and running
+
+    make
+    ./alertmanager -logtostderr -config.file=/path/to/alertmanager.conf
+
+## Configuring Prometheus to send alerts
+
+To make Prometheus send alerts to your Alertmanager, set the `alertmanager.url`
+command-line flag on the server:
+
+    ./prometheus -alertmanager.url=http://<alertmanager-host>:<port> <...other flags...>
+
+Prometheus only pushes firing alerts to Alertmanager. Alertmanager expects to
+receive regular pushes of firing alerts from Prometheus. Alerts which are not
+refreshed for a period of `-alerts.min-refresh-period` (5 minutes by
+default) are expired.
+
+Alertmanager only shows alerts which are currently firing and pushed to
+Alertmanager.
+
+## Running tests
+
+    make test
+
+## Caveats and roadmap
+
+Alertmanager is still in an experimental state. Some of the known caveats which
+are going to be addressed in the future:
+
+* Alertmanager is run as a single instance and does not provide high
+  availability yet. We plan on using a consensus protocol between multiple
```

Comment:
Paxos isn't appropriate for this sort of usage, as it can't handle partition. A single completely isolated alertmanager needs to be able to send alerts.

I'd have them all talk to each other, and choose a consistent leader (e.g. lowest IP). We'd then need some way to synchronise notification and silence state across them, likely using CRDTs so we get sane eventual consistency.